modules/casper.js

```diff
@@ -821,16 +821,19 @@ Casper.prototype.fillForm = function fillForm(selector, vals, options) {
                 if (!file || !file.path) {
                     return;
                 }
-                if (!fs.exists(file.path)) {
-                    throw new CasperError('Cannot upload nonexistent file: ' + file.path);
-                }
+                var paths = (utils.isArray(file.path) && file.path.length > 1) ? file.path :[file.path];
```

Comment:
When `file.path` is array with one or zero items, the `paths` will be an array containing an array.